docs/spec/fastsync/fastsync.md

```diff
@@ -0,0 +1,1164 @@
+Fastsync should bring a node within this range.
+
+### Blockchain
+
+> We will briefly list some of the notions
+> of blockchains as required for this specification. More details can
+> be found  [here][block].
+
+#### **[TMBC-HEADER]**:
+A set of blockchain transactions is stored in a data structure called
+*block*, which contains a field called *header*. (The data structure
+*block* is defined [here][block]).  As the header contains hashes to
+the relevant fields of the block, for the purpose of this
+specification, we will assume that the blockchain is a list of
+headers, rather than a list of blocks. 
+
+#### **[TMBC-SEQ]**:
+
+The Tendermint blockchain is a list *chain* of headers. 
+
+#### **[TMBC-SEQ-GROW]**: 
+
+During operation, new headers may be appended to the list one by one.
+
+
+> In the following, *ETIME* is a lower bound
+> on the time interval between the times at which two
+> successor blocks are added. 
+
+#### **[TMBC-SEQ-APPEND-E]**: 
+If a header is appended at time *t* then no additional header will be
+appended before time *t + ETIME*.
+
+
+#### **[TMBC-AUTH-BYZ]**:
+The authenticated Byzantine model assumes that no node (faulty or
+correct) may break digital signatures, but otherwise, no additional
+assumption is made about the internal behavior of faulty full
+nodes. That is, faulty nodes are only limited in that they cannot forge
+messages.
+
+<!-- The authenticated Byzantine model assumes [TMBC-Sign-NoForge] and -->
+<!-- [TMBC-FaultyFull], that is, faulty nodes are limited in that they -->
+<!-- cannot forge messages [TMBC-Sign-NoForge]. -->
+
+
+> We observed that in the existing documentation the term
+> *validator* refers to both a data structure and a full node that
+> participates in the distributed computation. Therefore, we introduce
+> the notions *validator pair* and *validator node*, respectively, to
+> distinguish these notions in the cases where they are not clear from
+> the context.
+
+
+#### **[TMBC-VALIDATOR-PAIR]**:
+
+Given a full node, a 
+*validator pair* is a pair *(address, voting_power)*, where 
+  - *address* is the address (public key) of a full node, 
+  - *voting_power* is an integer (representing the full node's
+  voting power in a certain consensus instance).
+  
+> In the Golang implementation the data type for *validator
+> pair* is called `Validator`
+
+#### **[TMBC-VALIDATOR-SET]**:
+
+A *validator set* is a set of validator pairs. For a validator set
+*vs*, we write *TotalVotingPower(vs)* for the sum of the voting powers
+of its validator pairs.
+
+#### **[TMBC-CORRECT]**:
+We define a predicate *correctUntil(n, t)*, where *n* is a node and *t* is a 
+time point. 
+The predicate *correctUntil(n, t)* is true if and only if the node *n* 
+follows all the protocols (at least) until time *t*.
+
+#### **[TMBC-TIME-PARAMS]**:
+A blockchain has the following configuration parameters:
+ - *unbondingPeriod*: a time duration.
```

Comment:
It seems that we don't use unbonding period term.

Reply:
I think we should still use it as people know the term?

Reply:
Ok, fair enough.